addons/gamification/models/res_users.py

```diff
@@ -290,10 +290,9 @@ def _get_next_rank(self):
 
         if self.next_rank_id:
             return self.next_rank_id
-        elif not self.rank_id:
-            return self.env['gamification.karma.rank'].search([], order="karma_min ASC", limit=1)
         else:
-            return self.env['gamification.karma.rank']
+            domain = [('karma_min', '>', self.rank_id.karma_min)] if self.rank_id else []
```

Comment:
can you write a small test ? (with some rank with karma lower, to be sure the domain is applied)

Reply:
Hello @std-odoo, thanks for the review. I have added the test in odoo/odoo#157585 (as the branch 16.1 has been disabled).